codepy/jit.py

```diff
@@ -230,7 +230,7 @@ def compile_from_string(toolchain, name, source_string,
     """
 
     # first ensure that source strings and names are lists
-    if isinstance(source_string, str):
+    if isinstance(source_string, str) or isinstance(source_string, bytes):
```

Comment:
Just depend on `six` for this. I imagine you're looking for `six.string_types`.

Don't forget to add the `six` dep to `setup.py`.

http://pythonhosted.org/six/

Reply:
Minor note: I wanted something along the lines of "if it's a string or if we're building from binary and it's a binary", see 5ffe811